### PR TITLE
#167891578 Implement Reset Password Feature

### PR DESCRIPTION
**Pull request by max-wel**
#### What does this PR do?
Implements reset password via email feature
#### Description of Task to be completed?
* Write tests
* Write email and password validations
* Create forgot password and reset password controllers
* Create helper function to send email with reset token to user
* Create forgot password and reset password routes
#### How should this be manually tested?
* Pull this branch ```ft-reset-password-167891578```
* Run ```npm run devstart``` command in the terminal
* Seed user into the database
* On postman, make a POST request to ```localhost:3000/api/v1/auth/forgot_password```
* Check the inbox of the user email you seeded into the database
* Get the reset token from the reset link sent and make a POST request with valid credentials to ```localhost:3000/api/v1/auth/reset_password/:{resetToken}```
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#167891578](https://www.pivotaltracker.com/story/show/167891578)
#### Screenshots (if appropriate)
Screenshot (123)

Screenshot (122)

### CODE REVIEW

File: src/database/migrations/20190822135538-company.js [hunk 1]
```diff
@@ -36,4 +35,4 @@ module.exports = {
     }
   })),
   down: queryInterface => queryInterface.dropTable('Companies')
-};
+};
```
Comment (thatkazuk1) [inline src/database/migrations/20190822135538-company.js]: Add newline here @sneakymaxy

**Review by femitj — APPROVED** [review]

File: src/controllers/Auth.js [hunk 2]
```diff
@@ -49,6 +52,72 @@ class Auth {
...
+    // check if user exists
+    try {
+      const user = await User.findOne({ where: { id: userId || null } });
+      if (!user) {
+        const response = new Response(false, 404, 'User does not exist');
+        return res.status(response.code).json(response);
+      }
+      // get reset secret (use user's password as secret to make token one time use only)
+      const { password: secret } = user.dataValues;
+      jwt.verify(resetToken, secret);
```
Comment (tvpeter) [inline src/controllers/Auth.js:198]: Seems you did not `await` the operation.

**Review by tvpeter — APPROVED** [review]
Massive work.

**Review by tvpeter — APPROVED** [review]
Seems your documentation style doesn't tally with others.

**Review by thatkazuk1 — APPROVED** [review]

File: src/controllers/Auth.js (hunk 2 above)
Comment (max-wel) [reply src/controllers/Auth.js:198]: jwt.verify() does not return a promise

**Review by Pomile — APPROVED** [review]
Nice Job!

**Review by daniellamarr — APPROVED** [review]
Good to go!

**Pull request merged**